fish-rust/src/builtins/shared.rs

```diff
@@ -79,7 +112,8 @@
     streams: Pin<&mut builtins_ffi::io_streams_t>,
     cpp_args: &Vec<wcharz_t>,
     builtin: RustBuiltin,
-) {
+    status_code: &mut i32,
```

Comment:
nit: CXX doesn't support type aliases in the extern declarations, but in the definition we can use `libc::c_int` (but I'm not sure if this is the perfect type, so feel free to ignore)